requests_toolbelt/utils/formdata.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+"""Implementation of nested form-data encoding function(s)."""
+from .._compat import basestring
+from .._compat import urlencode as _urlencode
+
+
+__all__ = ('urlencode',)
+
+
+def urlencode(query, doseq=False, safe='', encoding=None, errors=None):
+    """Enhanced urlencode method for handling nested queries.
+
+    When using this to send data in the body of a request, make sure you
+    specify the appropriate Content-Type header for the request.
+
+    .. code-block::
+
+        import requests
+        from requests_toolbelt.utils import formdata
+
+        query = {
+           'my_dict': {
+               'foo': 'bar',
+               'biz': 'baz",
+            },
+            'a': 'b',
+        }
+
+        resp = requests.get(url, params=formdata.urlencode(query))
+        # or
+        resp = requests.post(
+            url,
+            data=formdata.urlencode(query),
+            headers={
+                'Content-Type': 'application/x-www-form-urlencoded'
+            },
+        )
+
+    For parameter and return information, see the official `urlencode`_
+    documentation. (On Python 2, the ``safe``, ``encoding``, and ``errors``
+    parameters are ignored since they are not passed to the original
+    ``urllib.urlencode`` function.
+
+    .. _urlencode:
+        https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode
+    """
+    expand_classes = (dict, list, tuple)
+    original_query_list = _to_kv_list(query)
+
+    if not all(len(i) for i in original_query_list):
+        raise ValueError("Expected items to have a length of two.")
```

Comment:
Let me know if the new message is better. I jsut realized why the check is confusing. Coffee is finally kicking in.